### PR TITLE
[algoliasearch_v3.x.x] allow host to be string or array of string

### PR DESCRIPTION
**Pull request by jansalwowski**
- Links to documentation: https://github.com/algolia/algoliasearch-client-javascript#client-options
- Link to GitHub or NPM: https://github.com/algolia/algoliasearch-client-javascript/blob/develop/src/AlgoliaSearchCore.js#L100
- Type of contribution: fix

Other notes:
Algolia can receive array of hosts for read/write

### CODE REVIEW

File: definitions/npm/algoliasearch_v3.x.x/flow_v0.71.x-/algoliasearch_v3.x.x.js [hunk 1]
```diff
@@ -64,8 +64,8 @@ declare type $algoliasearch$ClientOptions = {|
   protocol?: "http:" | "https:",
   httpClient?: HTTPClient,
   hosts?: {|
-    read?: string,
-    write?: string
+    read?: string | Array<string>,
+    write?: string | Array<string>
```
Comment (pascalduez) [inline definitions/npm/algoliasearch_v3.x.x/flow_v0.71.x-/algoliasearch_v3.x.x.js]: Looks like it should be only `Array<string>`.    
Could you add/update tests in consequence?

Comment (jansalwowski) [reply definitions/npm/algoliasearch_v3.x.x/flow_v0.71.x-/algoliasearch_v3.x.x.js]: Algolia accepts borh string and array of strings. I will try to add tests, yet i have never tested flow typings before.

Comment (pascalduez) [reply definitions/npm/algoliasearch_v3.x.x/flow_v0.71.x-/algoliasearch_v3.x.x.js]: All the docs, unit test, even the TypeScript definitions allows only `Array<string>`. 
Also [sources](https://github.com/algolia/algoliasearch-client-javascript/blob/ba1f66222acb381372850da1d7caa39fd3c67580/src/AlgoliaSearchCore.js#L60-L63).

**Comment by jansalwowski** [comment]
@pascalduez I fixed typing and added a test for it, is it okay right now?

**Comment by pascalduez** [comment]
Thanks @jansalwowski!

**Pull request merged**